workers/tests/test_barrier.py

```
@@ -358,103 +402,568 @@ def test_create_chord_execution_delegates_to_module_barrier(self):
+    """
+
+    def test_manual_review_required_invokes_decision_helper_per_batch(
+        self, monkeypatch
+    ):
+        """With ``review_required=True``, the decision helper is
+        invoked exactly once per batch and the chord still fires
+        with the same fairness slot as the non-review path.
+
+        Drives the helper with ``num_batches=3`` so the "once per
+        batch" contract is genuinely exercised — a regression
+        invoking the helper 0× or 2× per batch wouldn't be caught
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)